File: _config/shortcodes/paired/update.js
import njk from "../../libraries/nunjucks.js";
import md from "../../libraries/markdown.js";

export default function (content, date) {
    return njk.renderString(
        '<aside class="update">' +
        '    <h5>Update {{ date }}:</h5>' +
        '    {{ content | safe }}' +
        '</aside>',
        {
            date,
            content: md.render(content.trim()),
        },
    );
};

File: cmd/serve/content/shortcodes/update.go
package shortcodes

import (
	"fmt"

	"chameth.com/chameth.com/cmd/serve/content/markdown"
	"chameth.com/chameth.com/cmd/serve/content/shortcodes/templates"
)

func renderUpdate(args []string, _ *Context) (string, error) {
	if len(args) < 2 {
		return "", fmt.Errorf("update requires at least 2 arguments (date, content)")
	}

	date := args[0]
	content := args[1]

	md, err := markdown.Render(content)
	if err != nil {
		return "", fmt.Errorf("failed to render update markdown: %w", err)
	}

	replacement, err := templates.RenderUpdate(templates.UpdateData{
		Date:    date,
		Content: md,
	})
	if err != nil {
		return "", fmt.Errorf("failed to render update template: %w", err)
	}

	return replacement, nil
}
